test/e2e/framework/util.go

```diff
@@ -349,6 +432,51 @@ func GetReadySchedulableNodesOrDie(c clientset.Interface) (nodes *v1.NodeList) {
 	return nodes
 }
 
+// newSecretTemplate returns the default v1.Secret template for this jig, but
+// does not actually create the Secret.
+func newSecretData(ns, generateName string, caCert, cert, privateKey, passphrase []byte) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      generateName,
+		},
+		Data: map[string][]byte{
+			cloudprovider.SSLCAFileName:          caCert,
+			cloudprovider.SSLCertificateFileName: cert,
+			cloudprovider.SSLPrivateKeyFileName:  privateKey,
+			cloudprovider.SSLPassphrase:          passphrase,
+		},
+	}
+}
+
+// CreateSSLSecretOrFail creates an SSL Kubernetes secret with the data passed in as
+// parameters
+func CreateSSLSecretOrFail(client clientset.Interface, ns, generateName string, caCert, cert, privateKey, passphrase []byte) error {
+	Logf("Creating SSL secret in ns %s", ns)
+	secret := newSecretData(ns, generateName, caCert, cert, privateKey, passphrase)
+	created, err := client.CoreV1().Secrets(ns).Create(secret)
+	Expect(err).NotTo(HaveOccurred())
+	Logf("Check if secret %s has been generated in ns %s", created.Name, secret.Namespace)
+	err = wait.PollImmediate(Poll, 5*time.Minute, func() (bool, error) {
```

Comment:
We can rely on Kubernetes to have created the Secret if it returns no error from Create so polling not required here AFAIK.